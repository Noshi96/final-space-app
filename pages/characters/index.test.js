/* globals describe, it, expect  */
/* eslint-env jest */

import { render, waitFor } from '@testing-library/react'
import Characters from './index.page'

describe('Characters', () => {
  const newCharacters = [
    {
      gender: 'Female',
      id: '01GFS938JAPCK8ZW24J43XA4AD',
      img_url:
        'https://images.pexels.com/photos/45201/kitty-cat-kâ€¦jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
      name: 'Catie',
      occurrence: '2',
      origin: 'Cat planet',
      species: 'Cat',
      status: 'Alive'
    }
  ]
  const characters = [
    {
      id: 1,
      name: 'Gary Goodspeed',
      status: 'Alive',
      species: 'Human',
      gender: 'Male',
      hair: 'Blonde',
      origin: 'Earth',
      abilities: [
        'Piloting',
        'Marksmanship',
        'Hand-to-hand combat',
        'Weapons: Blasters',
        'Weapons: Black hole gun (briefly)',
        'Weapons: Robot arm',
        'Weapons: Laser gun',
        'Weapons: Laser sword'
      ],
      img_url:
        'https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png'
    }
  ]

  const episodes = [
    {
      id: 1,
      name: 'Chapter 1',
      air_date: '02/15/2018',
      director: 'Mike Roberts',
      writer: 'Olan Rogers',
      characters: [
        'https://finalspaceapi.com/api/v0/character/1',
        'https://finalspaceapi.com/api/v0/character/2',
        'https://finalspaceapi.com/api/v0/character/3',
        'https://finalspaceapi.com/api/v0/character/5',
        'https://finalspaceapi.com/api/v0/character/6',
        'https://finalspaceapi.com/api/v0/character/9',
        'https://finalspaceapi.com/api/v0/character/11'
      ],
      img_url: 'http://finalspaceapi.com/api/episode/image/chapter1.jpg'
    }
  ]

  it('should render Characters page', async () => {
    const { container } = render(
      <Characters
        characters={characters}
        episodes={episodes}
        newCharacters={newCharacters}
      />
    )
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
