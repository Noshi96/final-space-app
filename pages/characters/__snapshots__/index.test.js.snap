// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should render Characters page 1`] = `
<div>
  <form
    class="sc-a77d193a-0 gnEPue"
  >
    <div
      class="MuiFormControl-root MuiTextField-root text css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="search-bar"
        id="search-bar-label"
      >
        Enter a character name
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id="search-bar"
          placeholder="Search..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Enter a character name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
  <button>
    Add Character
  </button>
  <div
    class="sc-da2bec26-0 cIXBnP"
  >
    <div
      class="sc-d6cb549b-0 jisOga"
      tabindex="0"
    >
      <img
        alt="Gary Goodspeed"
        class="sc-d6cb549b-2 cVwxUm"
        src="https://finalspaceapi.com/api/character/avatar/gary_goodspeed.png"
      />
      <div
        class="sc-d6cb549b-1 dfxNAZ"
      >
        Gary Goodspeed
      </div>
    </div>
  </div>
</div>
`;
